File: domain/src/main/java/com/omvp/domain/interactor/impl/RegisterDeviceUseCaseImpl.java
package com.omvp.domain.interactor.impl;

import com.omvp.domain.interactor.RegisterDeviceUseCase;
import com.omvp.domain.repository.DeviceRepository;

import javax.inject.Inject;

import io.reactivex.Completable;

public class RegisterDeviceUseCaseImpl extends BaseUseCaseImpl<DeviceRepository> implements RegisterDeviceUseCase {

    @Inject
    RegisterDeviceUseCaseImpl(DeviceRepository repository) {
        super(repository);
    }

    @Override
    public Completable execute() {
        return getRepository().register();
    }

}

File: domain/src/main/java/com/omvp/domain/interactor/impl/RegisterDeviceUseCaseImpl.kt
package com.omvp.domain.interactor.impl

import com.omvp.domain.interactor.RegisterDeviceUseCase
import com.omvp.domain.repository.DeviceRepository

import javax.inject.Inject

import io.reactivex.Completable

class RegisterDeviceUseCaseImpl @Inject
internal constructor(repository: DeviceRepository) : BaseUseCaseImpl<DeviceRepository>(repository), RegisterDeviceUseCase {

    override fun execute(): Completable {
        return repository.register()
    }

}
